server/auth.go

```diff
@@ -200,6 +220,56 @@ func (s *Server) registerAuthRoutes(g *echo.Group) {
 		}
 		return nil
 	})
+
+	// TODO(zilong): we may not need to return access token back to the frontend
+	g.POST("/auth/exchange-oauth-token/:vcsID", func(c echo.Context) error {
```

Comment:
let's have a separate oauth endpoint

/oauth/exchange-token/vcs/:vcsID

oauth is different from auth, and we may have other oauth scenarios other than vcs